litmus-portal/graphql-server/pkg/rbac/validate.go

```diff
@@ -0,0 +1,28 @@
+package validate
+
+import (
+	"context"
+	"errors"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	dbOperationsProject "github.com/litmuschaos/litmus/litmus-portal/graphql-server/pkg/database/mongodb/project"
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/litmuschaos/litmus/litmus-portal/graphql-server/graph/model"
+	"github.com/litmuschaos/litmus/litmus-portal/graphql-server/pkg/authorization"
+)
+
+// ValidateRole :Validates the role of a user in a given project
+func ValidateRole(ctx context.Context, projectID string, requiredRoles []model.MemberRole) error {
+	claims := ctx.Value(authorization.UserClaim).(jwt.MapClaims)
+	uid := claims["uid"].(string)
+
+	filter := bson.D{{"members", bson.D{{"$elemMatch", bson.D{{"user_id", uid}, {"role", bson.D{{"$in", requiredRoles}}}}}}}, {"_id", projectID}}
```

Comment:
Need to check for user invitation status, else exited or pending users will be able to do operations too.